src/sentry/tasks/relay.py

```diff
@@ -241,18 +241,50 @@ def invalidate_project_config(
 
 
 def schedule_invalidate_project_config(
-    *, trigger, organization_id=None, project_id=None, public_key=None
+    *,
+    trigger,
+    organization_id=None,
+    project_id=None,
+    public_key=None,
+    countdown=5,
 ):
     """Schedules the :func:`invalidate_project_config` task.
 
     This takes care of not scheduling a duplicate task if one is already scheduled.  The
     parameters are passed straight to the task.
+
+    :param countdown: The time to delay running this task in seconds.  Normally this is
+       delay a little to increase the likelyhood of deduplicating invalidations but you can
+       tweak this, like e.g. the :func:`invalidate_all` task does.
     """
+    from sentry.models import Project, ProjectKey
+
     validate_args(organization_id, project_id, public_key)
 
-    if projectconfig_debounce_cache.invalidation.is_debounced(
-        public_key=public_key, project_id=project_id, organization_id=organization_id
-    ):
+    # The keys we need to check for to see if this is debounced, we want to check all
+    # levels.
+    check_debounce_keys = {
+        "public_key": public_key,
+        "project_id": project_id,
+        "organization_id": organization_id,
+    }
+    if public_key:
+        try:
+            key = ProjectKey.objects.get(public_key=public_key)
+        except ProjectKey.DoesNotExist:
+            pass
+        else:
+            check_debounce_keys["project_id"] = key.project.id
+            check_debounce_keys["organization_id"] = key.project.organization.id
+    elif project_id:
+        try:
+            proj = Project.objects.get(id=project_id)
+        except Project.DoesNotExist:
+            pass
+        else:
+            check_debounce_keys["organization_id"] = proj.organization.id
```

Comment:
Doesn't this mean that when an org invalidation is scheduled right after a single project of that org has been scheduled, the _other_ projects in that org will remain stale even though they should be updated?